internal/sdkv2provider/resource_cloudflare_api_shield_schema_validation_settings.go

```diff
@@ -0,0 +1,123 @@
+		Schema:        resourceCloudflareAPIShieldSchemaValidationSettingsSchema(),
+		CreateContext: resourceCloudflareAPIShieldSchemaValidationSettingsCreate,
+		ReadContext:   resourceCloudflareAPIShieldSchemaValidationSettingsRead,
+		UpdateContext: resourceCloudflareAPIShieldSchemaValidationSettingsUpdate,
+		DeleteContext: resourceCloudflareAPIShieldSchemaValidationSettingsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: nil,
+		},
+		Description: heredoc.Doc(`
+			Provides a resource to manage settings in API Shield Schema Validation 2.0.
+		`),
+	}
+}
+
+func resourceCloudflareAPIShieldSchemaValidationSettingsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceCloudflareAPIShieldSchemaValidationSettingsUpdate(ctx, d, meta)
+}
+
+func resourceCloudflareAPIShieldSchemaValidationSettingsUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get(consts.ZoneIDSchemaKey).(string)
+
+	dm := d.Get("validation_default_mitigation_action").(string)
+	om := d.Get("validation_override_mitigation_action").(string)
+
+	_, err := client.UpdateAPIShieldSchemaValidationSettings(
+		ctx,
+		cloudflare.ZoneIdentifier(zoneID),
+		cloudflare.UpdateAPIShieldSchemaValidationSettingsParams{
+			DefaultMitigationAction:  &dm,
+			OverrideMitigationAction: &om,
+		},
+	)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "failed to create API Shield Schema Validation Settings"))
+	}
+
+	// Settings are configured at the zone level so using the zoneID as the ID
+	d.SetId(zoneID)
+	return resourceCloudflareAPIShieldSchemaValidationSettingsRead(ctx, d, meta)
+}
+
+func resourceCloudflareAPIShieldSchemaValidationSettingsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get(consts.ZoneIDSchemaKey).(string)
+
+	settings, err := client.GetAPIShieldSchemaValidationSettings(
+		ctx,
+		cloudflare.ZoneIdentifier(zoneID),
+	)
+
+	defaultSettings := cloudflareAPIShieldSchemaValidationSettingsDefault()
+
+	// translate nil to the default setting
+	if settings.OverrideMitigationAction == nil {
+		settings.OverrideMitigationAction = defaultSettings.OverrideMitigationAction
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to fetch API Shield Schema Validation Settings: %w", err))
+	}
+
+	if err := d.Set("validation_default_mitigation_action", settings.DefaultMitigationAction); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("validation_override_mitigation_action", *settings.OverrideMitigationAction); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Settings are configured at the zone level so using the zoneID as the ID
+	d.SetId(zoneID)
+	return nil
+}
+
+func resourceCloudflareAPIShieldSchemaValidationSettingsDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Why is this a delete? We are not deleting anything? Or do you have to give a delete-function here?

Reply:
if a resource is removed from the terraform state, it deletes the resource - as we don't actually "delete" anything this just resets the state of the settings to default. 

It could be argued there should be a `DELETE` endpoint on the API, however this implementation will achieve the same.